src/libcore/str/mod.rs

```diff
@@ -1150,7 +1150,7 @@ mod traits {
 
 /// Any string that can be represented as a slice
 #[unstable = "Instead of taking this bound generically, this trait will be \
-              replaced with one of slicing syntax, deref coercions, or \
+              replaced with one of slicing syntax (&foo[]), deref coercions (&*foo), or \
```

Comment:
Perhaps this could avoid using `(&*foo)` explicitly? Eventually it'll in theory just be `&foo` to coerce to `&str` from `&String`.